Google.Api.Gax.Grpc/ApiClientStreamingCall.cs

```diff
@@ -0,0 +1,73 @@
+﻿/*
+ * Copyright 2021 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using Grpc.Core;
+using System;
+
+namespace Google.Api.Gax.Grpc
+{
+    internal static class ApiClientStreamingCall
+    {
+        internal static ApiClientStreamingCall<TRequest, TResponse> Create<TRequest, TResponse>(
+            Func<CallOptions, AsyncClientStreamingCall<TRequest, TResponse>> grpcCall,
+            CallSettings baseCallSettings,
+            ClientStreamingSettings streamingSettings,
+            IClock clock) =>
+                new ApiClientStreamingCall<TRequest, TResponse>(
+                    cs => grpcCall(cs.ValidateNoRetry().ToCallOptions(clock)),
+                    baseCallSettings,
+                    streamingSettings);
+    }
+
+    /// <summary>
+    /// Bridge between a client streaming RPC method and higher level
+    /// abstractions, applying call settings as required.
+    /// </summary>
+    /// <typeparam name="TRequest"></typeparam>
```

Comment:
Sorry, forgot to mention this before - please add docs for these type parameters.

Reply:
👍 Thank you for catching these.